### ISSUE TITLE
make %d directive human readable

### ISSUE THREAD

**Issue by rsgalloway** [rsgalloway/pyseq#20]
Currently using the %d directive in seq formatting, the displayed values are in bytes. It'd be nice if this was in a human readable format, e.g. 48KB, or 5MB, etc. 

Or, add a new %D directive for this and leave %d in bytes.

**Comment by broganross** [comment]
Not sure if this is valid but don't forget that the size returned from getsize, os.stat().st_size, is not the same thing as the size taken up on disk.

**Comment by rsgalloway** [comment]
Really? Can you enlighten me? Seems to be the common recommendation for getting file size in bytes. Also, did a quick test, and seemed to return the right value.

**Comment by broganross** [comment]
So os.stat().st_size gives you back the size of the contents of the file in bytes.  But there's also the size that a storage system allocates to a file based on a block size .  Basically, the storage device will allocate space for a file not only based on  it's current size, but it's size rounded up to the nearest multiple of it's block size (os.stat().st_blksize).  So if you have a file who's contents is 2 bytes, on a file system who's block size is 512KB, then the file system is actually reserving 512KB of space on the drive for that file.  This is why, at least on some *nix machines, you get different size results from ls and du.  Or du and adding up the sizes with st_size in python.
It may not seem like that big a deal but if you have directories with a lot of files those extra bytes will start to add up.

Basically if you want to calculate the size to blocksize of a file, it's os.stat().st_size rounded to the nearest multiple of os.stat().st_blksize.  There's methods out there for doing a ceiling to a multiple.  If you're doing it for directory, round each file size and add them.
Hopefully, that makes sense.  And like I said I don't know if this is valid here, but I've had to deal with it when handling large numbers of files between different devices.

**Comment by rsgalloway** [comment]
Trying to close out a few issues, so coming back to this: I'd like to keep %d backwards compatible. So, we could add a new directive (%h is taken so maybe %H or %u?), or see if it can live as an option on lss.

**Comment by rsgalloway** [comment]
sounds good

**Comment by rsgalloway** [comment]
this feature was implemented in 0.7.0
